studio/src/components/app-provider.tsx

```diff
@@ -93,6 +99,10 @@ const fetchSession = async () => {
     }
     return null;
   } catch (e) {
+    // Reset koala if custom head scripts are found and user is not authenticated
+    if (process.env.CUSTOM_HEAD_SCRIPTS) {
+      window.ko?.reset();
```

Comment:
We also need to call `reset` when the user clicks on logout.

Reply:
Done.